src/Jackalope/Node.php

```diff
@@ -302,7 +310,9 @@ private function parseData($rawData, $update, $keepChanges = false)
                 $this->nodes = $nodesInBackend;
                 $this->originalNodesOrder = null;
             }
-            foreach ($oldProperties as $name => $property) {
+            foreach ($oldPropertyValues as $name => $propertyValues) {
+                // TODO needs an update
+                $property = $this->loadProperty($name);
```

Comment:
i think you try to load the new property here, where you should load the old data.